tests/types/npm-test.json

```diff
@@ -254,6 +254,316 @@
+      "input": "pkg:npm/@angular/animation@12.3.1",
+      "expected_output": null,
+      "expected_failure": true,
+      "expected_failure_reason": "Should fail to parse due to unencoded '@' in the namespace"
+    },
+    {
+      "description": "Negative Test: An npm purl with an uppercase name should fail parsing",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/React@17.0.1",
+      "expected_output": null,
+      "expected_failure": true,
+      "expected_failure_reason": "Should fail to parse due to uppercase characters in the name"
+    },
+    {
+      "description": "Normalization Test: An npm purl with an uppercase name should be lowercased on roundtrip",
+      "test_group": "advanced",
+      "test_type": "roundtrip",
+      "input": "pkg:npm/LoDash@4.17.21",
+      "expected_output": "pkg:npm/lodash@4.17.21",
+      "expected_failure": false,
+      "expected_failure_reason": null
+    },
+    {
+      "description": "Positive Test: An npm purl with no version is valid",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/react",
+      "expected_output": {
+        "type": "npm",
+        "namespace": null,
+        "name": "react",
+        "version": null,
+        "qualifiers": null,
+        "subpath": null
+      },
+      "expected_failure": false,
+      "expected_failure_reason": null
+    },
+    {
+      "description": "Positive Test: A scoped npm purl with no version is valid",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/%40angular/core",
+      "expected_output": {
+        "type": "npm",
+        "namespace": "@angular",
+        "name": "core",
+        "version": null,
+        "qualifiers": null,
+        "subpath": null
+      },
+      "expected_failure": false,
+      "expected_failure_reason": null
+    },
+    {
+      "description": "Negative Test: An npm purl with an empty version should fail",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/foobar@",
+      "expected_output": null,
+      "expected_failure": true,
+      "expected_failure_reason": "Should fail to parse due to an empty version string"
+    },
+    {
+      "description": "Negative Test: An npm purl with a name starting with a dot should fail",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/.foobar@1.0.0",
+      "expected_output": null,
+      "expected_failure": true,
+      "expected_failure_reason": "Should fail to parse because the name starts with a dot"
+    },
+    {
+      "description": "Negative Test: An npm purl with a name starting with an underscore should fail",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/_foobar@1.0.0",
+      "expected_output": null,
+      "expected_failure": true,
+      "expected_failure_reason": "Should fail to parse because the name starts with an underscore"
+    },
```

Comment:
Why should these fail? PURL does not (and IMO should not) define any rules to forbid this.